tiledb/sm/c_api/tiledb.cc

```diff
@@ -3718,6 +3718,60 @@ int32_t tiledb_array_consolidate_metadata_with_key(
   return TILEDB_OK;
 }
 
+int32_t tiledb_array_get_fragments(
```

Comment:
There is a lot of logic in this function. We decided previously to try to avoid putting login into the c-api function and instead keep them in the core c++ classes/files. This allow better reuse and for calling functions in the serialization code paths when needed.

Reply:
I agree with keeping the C-API thin, but this is entirely converting from an `std::vector<URI>` to a `char**`. This is in-line with the existing logic in this layer, it's just more verbose because of the double-pointer memory management. Do we really want to put this c-style stuff in the C++ StorageManager?